lib/api/dsl/methods.js

```diff
@@ -856,11 +858,12 @@ function buildCurriedFunction(index, collection, field, operatorName, value, cur
   }
 
   if (!methods.dsl.filtersTree[index][collection].fields[field][curriedFunctionName]) {
-    curriedFunction = _.curry(operators[operatorName]);
-    curriedFunction = _.curry(curriedFunction(field, value));
-    if (not) {
-      curriedFunction = _.negate(curriedFunction);
-    }
+    // we need the ugly eval code to make sure the function can later on be serialized and
```

Comment:
For the record, here is what we decided Benoit and I about this matter: we'll try to get rid of this `eval` call by only storing operators arguments instead of a currified function. 
This should be better performance-wise (because we would get rid of eval + 2 x JSON.stringify), but we don't know how it would perform memory-wise. Should be way better than the situation before this pull request.

The operators arguments will be as easily shareable across all kuzzle instances than what Benoit is proposing in this pull request.

I'll check if that's feasible in a reasonable amount of time (according to Benoit, it should be), in another branch starting from this one.